warehouse/warehouse/views.py

```diff
@@ -1,17 +1,41 @@
+import logging
+import random
+import time
+from concurrent.futures import ThreadPoolExecutor
 from rest_framework.response import Response
 from rest_framework.decorators import api_view
-import logging
+from django.conf import settings
 
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
+tracing = settings.OPENTRACING_TRACING
+tracer = tracing.tracer
+
 
 @api_view(http_method_names=["GET"])
 def fetch(request, order_num):
+    time.sleep(1)
+    if random.randint(1, 8) == 8:
+        msg = "Random Service Unavailable!"
+        logging.warning(msg)
+        return Response(msg, status=503)
     if not order_num:
         msg = "Invalid Order Num!"
         logging.warning(msg)
         return Response(msg, status=400)
+    executor = ThreadPoolExecutor(max_workers=2)
```

Comment:
Should this be defined outside the method? We don't want to create a executor every time we get a request? We just want to submit tasks to an existing executor.